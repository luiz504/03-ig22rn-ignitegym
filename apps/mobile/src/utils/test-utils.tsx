import { NavigationContainer } from '@react-navigation/native'
import { render } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import { FC, ReactNode } from 'react'
import { THEME } from '~/theme'

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  const inset = {
    frame: { x: 0, y: 0, width: 0, height: 0 },
    insets: { top: 0, left: 0, right: 0, bottom: 0 },
  }
  return (
    <NativeBaseProvider theme={THEME} initialWindowMetrics={inset}>
      <NavigationContainer>{children}</NavigationContainer>
    </NativeBaseProvider>
  )
}

const customRender: typeof render = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react-native'

export { customRender as render }
