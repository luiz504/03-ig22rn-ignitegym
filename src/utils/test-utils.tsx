import { render } from '@testing-library/react-native'
import { NativeBaseProvider } from 'native-base'
import { FC, ReactNode } from 'react'
import { THEME } from '~/theme'

const AllTheProviders: FC<{ children: ReactNode }> = ({ children }) => {
  return <NativeBaseProvider theme={THEME}>{children}</NativeBaseProvider>
}

const customRender: typeof render = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react-native'

export { customRender as render }
