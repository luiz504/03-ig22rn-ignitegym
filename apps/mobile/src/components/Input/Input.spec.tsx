import { renderWithNBProviders, screen } from '~/utils/test/test-utils'
import { Input } from '.'
import { Input as NBInput } from 'native-base'

describe('Input Component', () => {
  const inputID = 'input'
  const textErrorID = `${inputID}-error-text`
  it('should render Correctly', async () => {
    renderWithNBProviders(<Input testID={inputID} />)

    // Initial Assert
    const inputElement = screen.getByTestId(inputID)
    expect(inputElement.props.placeholderTextColor).toEqual('#7C7C8A')

    expect(inputElement.props.style).toMatchSnapshot(
      'initial-input-non-focused',
    )
    expect(screen.queryByTestId(textErrorID)).toBeNull()
  })

  it('should render correctly the errors and it layout effects', async () => {
    const textError = 'This is an error'
    renderWithNBProviders(<Input testID={inputID} errorMsg={textError} />)

    const inputElement = screen.UNSAFE_getByType(NBInput)
    const errorTextElement = screen.getByTestId(textErrorID)

    expect(inputElement.props.isInvalid).toBe(true)
    expect(errorTextElement).toHaveTextContent(textError)
  })
})
