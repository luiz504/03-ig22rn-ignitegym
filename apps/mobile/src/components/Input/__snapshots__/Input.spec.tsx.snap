// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Component should render Correctly: initial-input-non-focused 1`] = `
{
  "backgroundColor": "transparent",
  "color": "#FFFFFF",
  "flex": 1,
  "fontFamily": "Roboto-Regular",
  "fontSize": 16,
  "fontStyle": "normal",
  "fontWeight": "400",
  "height": "100%",
  "paddingBottom": 8,
  "paddingLeft": 16,
  "paddingRight": 16,
  "paddingTop": 8,
  "width": "100%",
}
`;
